### ISSUE TITLE
CONV f( 'abc' ) silently yields NaN instead of raising `cx_sy_conversion_no_number`

### ISSUE THREAD

**Issue by furkancosgun** [abaplint/transpiler#1782]
Converting a non-numeric string to type `f` does not raise `cx_sy_conversion_no_number` as real ABAP does. Instead the value silently becomes `NaN`, which then propagates through all subsequent arithmetic with no error.

### Minimal reproduction

```abap
TRY.
    DATA(result) = CONV f( 'not-a-number' ).
    WRITE result.   " prints nothing useful, value is NaN
  CATCH cx_sy_conversion_no_number.
    WRITE 'expected this branch'.
ENDTRY.
```
The CATCH branch is never reached.

Root cause
Float.set() in the runtime does this.value = parseFloat(value) (build/src/types/float.js:62) with no format validation. parseFloat('not-a-number') returns NaN, which is stored as-is.

This is inconsistent with Integer.set(), which validates the string against the DIGITS regex and calls throwError("CX_SY_CONVERSION_NO_NUMBER") when it does not match (build/src/types/integer.js:21-27).

There is a second, related problem: parseFloat('12abc') returns 12 — a partial parse is silently accepted, while real ABAP would also raise cx_sy_conversion_no_number for '12abc'.

Impact
Any code relying on TRY/CATCH cx_sy_conversion_no_number to detect non-numeric input silently breaks. In our project (an ABAP decision-tree library) the data loader uses CONV f( raw_value ) inside such a TRY/CATCH to reject non-numeric CSV features; the check never fired and NaN leaked into the training data. We had to add a workaround:

result = CONV f( feature_value ).
IF result <> result.   " NaN self-comparison, workaround for missing exception
  RAISE EXCEPTION TYPE zcx_dt_exception.
ENDIF.

Expected behavior
Float.set() should validate numeric strings the same way Integer.set() does and raise cx_sy_conversion_no_number (or the same runtime error) for non-numeric input, instead of storing NaN. At minimum, a string that fails parseFloat completely should not be accepted silently.

---

Additional minimal reproduction (full sample program):

```abap
REPORT zdt_demo_nan.

CLASS lcl_nan_demo DEFINITION.
  PUBLIC SECTION.
    CLASS-METHODS run.
  PRIVATE SECTION.
    CLASS-METHODS float_conversion.
    CLASS-METHODS integer_conversion.
    CLASS-METHODS arithmetic_propagation.
ENDCLASS.

CLASS lcl_nan_demo IMPLEMENTATION.
  METHOD run.
    WRITE / '=== 1) CONV f( ''abc'' ): should raise cx_sy_conversion_no_number ==='.
    float_conversion( ).

    WRITE / '=== 2) CONV i( ''abc'' ): reference behaviour (real ABAP) ==='.
    integer_conversion( ).

    WRITE / '=== 3) NaN propagates silently into arithmetic ==='.
    arithmetic_propagation( ).
  ENDMETHOD.

  METHOD float_conversion.
    DATA(raw) = 'abc'.
    TRY.
        DATA(value) = CONV f( raw ).
        WRITE / 'No exception raised!'.
        WRITE / |value = { value }|.
        WRITE / |NaN self-comparison ( value <> value ): { xsdbool( value <> value ) }|.
      CATCH cx_sy_conversion_no_number.
        WRITE / 'CATCH cx_sy_conversion_no_number reached (expected in real ABAP)'.
    ENDTRY.
  ENDMETHOD.

  METHOD integer_conversion.
    DATA(raw) = 'abc'.
    TRY.
        DATA(value) = CONV i( raw ).
        WRITE / |No exception raised, value = { value }|.
      CATCH cx_sy_conversion_no_number.
        WRITE / 'CATCH cx_sy_conversion_no_number reached (same as real ABAP)'.
    ENDTRY.
  ENDMETHOD.

  METHOD arithmetic_propagation.
    DATA(raw) = 'abc'.
    DATA(value) = CONV f( raw ).
    DATA(result) = value + 1.
    WRITE / |value + 1 = { result }|.
  ENDMETHOD.
ENDCLASS.

START-OF-SELECTION.
  lcl_nan_demo=>run( ).

```
